ui/prometheus-plugin/src/plugins/prometheus-time-series-query/PrometheusTimeSeriesQueryEditor.tsx

```diff
@@ -81,7 +81,8 @@ export function PrometheusTimeSeriesQueryEditor(props: PrometheusTimeSeriesQuery
       </FormControl>
       <PromQLEditor
         completeConfig={{ remote: { url: promURL } }}
-        value={query}
+        value={value.query}
```

Comment:
This change sounds to counter a mechanism of ``useQueryState``? can you explain? At a glance it seems this trick can introduce confusion.

Reply:
I want the tree view to be (re)computed when the input loses focus, and not after each change, to avoid spamming API calls - just like we do for the queries to retrieve the data. Hence the trick here, `value.query` gets updated only after `handleQueryBlur` from `useQueryState` is called, while the `query` object provided by `useQueryState` gets updated after each call to  `handleQueryChange`.

Reply:
Added comment for clarification.